sdks/python/apache_beam/utils/windowed_value.pxd

```diff
@@ -43,6 +43,14 @@ cdef class WindowedValue(object):
 
   cpdef WindowedValue with_value(self, new_value)
 
+@cython.final
```

Comment:
Let's make this an interface. (Maybe split it out into a different PR as well.)

Reply:
In particular, for non-1:1 batch processing, we may want to (optionally?) expose the windowing information as part of the batch itself, e.g. as additional columns (for a dataframe-like batch), or as a list of WindowedValues (for a List batch), or ???

An alternative is to have batches of homogeneous windowing information (which would allow dropping the 1:1 requirement, at the expense of possibly smaller batches).

Reply:
Ack I removed this from a PR. What does it mean to make it an interface? Just drop the `@cython.final`? (sorry my knowledge of cython is very limited :sweat_smile:)